aggregator/prover/prover.go

```diff
@@ -2,81 +2,254 @@ package prover
 	if err != nil {
-		log.Fatalf("fail to dial: %v", err)
+		return nil, err
 	}
+	p.id = status.ProverId
+	return p, nil
+}
 
-	proverClient := proverclientpb.NewZKProverServiceClient(proverConn)
-	prover := &Prover{URI: proverURI, Client: proverClient, Conn: proverConn}
+// ID returns the Prover ID.
+func (p *Prover) ID() string { return p.id }
 
-	go func() {
-		waitTick(ctx, tickerCheckConnection)
-		for {
-			prover.checkConnection(ctx, tickerCheckConnection)
-		}
-	}()
+// Status gets the prover status.
+func (p *Prover) Status() (*pb.GetStatusResponse, error) {
+	req := &pb.AggregatorMessage{
+		Request: &pb.AggregatorMessage_GetStatusRequest{
+			GetStatusRequest: &pb.GetStatusRequest{},
+		},
+	}
+	res, err := p.call(req)
+	if err != nil {
+		return nil, err
+	}
+	if msg, ok := res.Response.(*pb.ProverMessage_GetStatusResponse); ok {
+		return msg.GetStatusResponse, nil
+	}
+	return nil, errors.New("Bad response") // FIXME(pg)
+}
 
-	return prover
+// IsIdle returns true if the prover is idling.
+func (p *Prover) IsIdle() bool {
+	status, err := p.Status()
+	if err != nil {
+		log.Warnf("Error asking status for prover ID %s: %w", p.ID(), err)
+		return false
+	}
+	return status.Status == pb.GetStatusResponse_IDLE
 }
 
-func (p *Prover) checkConnection(ctx context.Context, ticker *time.Ticker) {
-	state := p.Conn.GetState()
-	log.Debugf("Checking connection to prover %v. State: %v", p.URI, state)
+// BatchProof instructs the prover to generate a batch proof for the provided
+// input. It returns the ID of the proof being computed.
+func (p *Prover) BatchProof(input *pb.InputProver) (string, error) {
+	req := &pb.AggregatorMessage{
+		Request: &pb.AggregatorMessage_GenBatchProofRequest{
+			GenBatchProofRequest: &pb.GenBatchProofRequest{Input: input},
+		},
+	}
+	res, err := p.call(req)
+	if err != nil {
+		return "", err
+	}
+	if msg, ok := res.Response.(*pb.ProverMessage_GenBatchProofResponse); ok {
+		switch msg.GenBatchProofResponse.Result {
+		case pb.Result_OK:
+			return msg.GenBatchProofResponse.Id, nil
+		case pb.Result_ERROR:
+			return "", errors.New("GenBatchProofResponse.Result: ERROR")
+		case pb.Result_INTERNAL_ERROR:
+			return "", errors.New("GenBatchProofResponse.Result: INTERNAL_ERROR")
+		case pb.Result_UNSPECIFIED:
+			return "", errors.New("GenBatchProofResponse.Result: UNSPECIFIED")
+		default:
+			return "", errors.New("GenBatchProofResponse.Result: UNKNOWN")
+		}
+	}
+
+	return "", errors.New("GenBatchProofResponse.Result: UNKNOWN")
+}
 
-	if state != connectivity.Ready {
-		log.Infof("Connection to prover %v seems broken. Trying to reconnect...", p.URI)
-		if err := p.Conn.Close(); err != nil {
-			log.Errorf("Could not properly close gRPC connection: %v", err)
+// AggregatedProof instructs the prover to generate an aggregated proof from
+// the two inputs provided. It returns the ID of the proof being computed.
+func (p *Prover) AggregatedProof(inputProof1, inputProof2 string) (string, error) {
```

Comment:
Why do you use strings instead of byte arrays? 

Again I think it is better to return *string.

Reply:
@ToniRamirezM see my answer above https://github.com/0xPolygonHermez/zkevm-node/commit/1252a0e9284de44e8ef5540e071f72dcb9f1544f